### PR TITLE
Support all SQL template libraries

### PR DESCRIPTION
**Pull request by blakeembrey**
Expanding the TypeScript interface to support anything that matches the expected object, not just the exact interface provided by `sql-template-strings`. I'm using [this file](https://github.com/kriasoft/node-sqlite/blob/master/src/utils/strings.ts) as the example for supported input types, which explicitly seems to be only caring about `values` as a property.

### CODE REVIEW

File: src/interfaces.ts [hunk 1]
```diff
@@ -11,6 +10,11 @@ export namespace ISqlite {
     params?: any[]
   }
 
+  export interface SQLStatement {
+    sql: string
+    values?: any[]
```
Comment (blakeembrey) [inline src/interfaces.ts:15]: I matched the existing code pattern (i.e. no semicolons, using `any`), let me know if you prefer `unknown` instead.

**Comment by theogravity** [comment]
This looks ok. I'll try testing it tonight, thanks!

**Comment by theogravity** [comment]
@blakeembrey sorry for the delay, just a lot of last-minute work-related requests on my end

looking at https://github.com/felixfbecker/node-sql-template-strings/blob/master/index.d.ts

there's a lot more going on than just the value field - it's an actual class with methods

concerned that not using it would break something

a potential compromise might be that we make all those items an interface instead

**Comment by blakeembrey** [comment]
I'm not 100% sure what you're worried about breaking so I'd love to clarify anything I can. Basically, it's all those extra fields that is causing the problem. In your code, you're only accessing these two fields: https://github.com/kriasoft/node-sqlite/blob/b7bae0dbc8d86d0490911b8e6b8bbee784d39de7/src/utils/strings.ts#L22-L23

So the correct interface is one that only accepts those two fields. None of the other methods are relevant to this library. For more context, I'd like to be using my own library which exposes those exact same fields and works for other libraries already like MySQL and Postgres: https://github.com/blakeembrey/sql-template-tag.

Since this is only the argument input type, and the JS only cares about those two fields, it's better to be agnostic to all the extra fields. This can also let someone just write `{ sql: '', values: [] }` if they really wanted to and have it "just work" because this library doesn't actually care in the end.

**Comment by blakeembrey** [comment]
I went ahead and removed the vendor version of `sql-template-strings` if that's reasonable to you, but it's still a part of the test suite and passing. Let me know if I can alleviate any concerns.

**Comment by theogravity** [comment]
Looks good, thanks for your patience!

**Review by theogravity — APPROVED** [review]

**Pull request merged**